service/build.gradle

```diff
@@ -1,138 +1,144 @@
-	implementation group: 'bio.terra.cloud-resource-lib', name: 'google-notebooks', version: '0.5.0'
-	implementation group: 'bio.terra.cloud-resource-lib', name: 'google-serviceusage', version: '0.7.0'
-	implementation group: 'bio.terra.cloud-resource-lib', name: 'google-storage', version: '0.8.0'
-	implementation group: 'bio.terra.cloud-resource-lib', name: 'google-iam', version: '0.7.0'
-	implementation group: "org.broadinstitute.dsde.workbench", name: "sam-client_2.12", version: "0.1-61135c7"
-	implementation group: "bio.terra", name: "terra-resource-buffer-client", version: "0.4.3-SNAPSHOT"
-
-	// Versioned direct deps
-	implementation group: "com.google.auto.value", name: "auto-value-annotations", version: "1.7.4"
-	implementation group: "com.google.guava", name:"guava", version: "30.0-jre"
-	implementation group: "org.hashids", name: "hashids", version: "1.0.3"
-	implementation group: "org.liquibase", name: "liquibase-core", version: "4.2.1"
-	implementation group: "org.webjars", name: "webjars-locator-core", version: "0.46"
-	runtimeOnly group: "org.postgresql", name: "postgresql", version: "42.2.18"
-
-	// Deps whose versions are controlled by Spring
-	implementation group: "javax.validation", name: "validation-api"
-	implementation group: "org.apache.commons", name: "commons-dbcp2"
-	implementation group: "org.apache.commons", name: "commons-lang3"
-	implementation group: "org.apache.commons", name: "commons-pool2"
-	implementation group: "org.springframework.boot", name: "spring-boot-starter-data-jdbc"
-	implementation group: "org.springframework.boot", name: "spring-boot-starter-web"
-
-	// OpenAPI (swagger) deps
-	implementation gradle.librarySwaggerAnnotations
-	swaggerCodegen gradle.librarySwaggerCli
-	runtimeOnly group: "org.webjars.npm", name: "swagger-ui-dist", version: "3.37.2"
-
-	// Test deps
-	testImplementation group: "com.google.auth", name: "google-auth-library-oauth2-http", version: "0.22.1"
-	testImplementation group: "io.vavr", name: "vavr", version: "0.10.3"
-	testImplementation group: "org.springframework.boot", name: "spring-boot-starter-test"
-	testImplementation project(":client")
-
-	annotationProcessor group: "com.google.auto.value", name: "auto-value", version: "1.7.4"
-	annotationProcessor group: "org.springframework.boot", name: "spring-boot-configuration-processor", version: "2.4.0"
+    // Google dependencies
+    constraints {
+        implementation 'com.google.guava:guava:30.1.1-jre' // "-jre" for Java 8 or higher
+    }
+    implementation platform('com.google.cloud:libraries-bom:20.2.0')
+    implementation group: "com.google.auto.value", name: "auto-value-annotations"
+    implementation group: "com.google.guava", name:"guava"
+    
+    // Terra deps
+    implementation group: "bio.terra", name: "datarepo-client", version: "1.41.0-SNAPSHOT"
+    // hk2 is required for datarepo client, but not correctly included
```

Comment:
How is it not correctly included? Can it not be fixed from here?

Reply:
datarepo should have `hk2` as an `api` dependency, since it is needed by users of the client. They have it as an `implementation` dependency so it doesn't show through. We have to include it ourselves.

Reply:
I improved the comment and added a ticket for the TDR project